Сетка страниц и крупных блоков на флексах

OlgaNikl opened the pull request:
---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/34/user/2067969/tasks/7)

:boom: https://htmlacademy-htmlcss.github.io/2067969-sedona-34/5/

OlgaNikl committed:
Исправила недочеты

исправила опечатку в container; добавила стили к main-container и inner-container;

OlgaNikl committed:
Сетка страниц и основных блоков

talequale commented on styles/styles.css:
```diff
@@ -190,6 +295,26 @@ body {
   text-transform: uppercase;
 }
 
+.subscribe-discription {
```
опечатка, правильно description

talequale commented on catalog.html:
```diff
@@ -8,8 +8,8 @@
 <body>
-  <div class="inner-conteiner">
-    <header>
+  <div class="inner-container">
```
это же одинаковые обёртки на обеих страницах, класс тоже должен быть одинаковым, чтоб не писать одни и те же стили дважды. сейчас вот такой дубль 

![image](https://user-images.githubusercontent.com/11165551/173199178-1579e05c-6de3-4dbe-b907-76a6aed2f399.png)

![image](https://user-images.githubusercontent.com/11165551/173199168-e07aa389-4b80-445f-8a7c-114596e73bdb.png)

talequale commented on styles/styles.css:
```diff
@@ -14,16 +14,27 @@
+  margin: 0;
+  display: flex;
+  flex-direction: column;
+  min-height: 100%;
```
min-width: 1200px также надо body указать, поскольку минимальная ширина контента такова, а body не может быть уже, чем контентная область

talequale commented on styles/styles.css:
```diff
@@ -67,13 +78,45 @@ body {
 }
 
+.page-header {
+  width: 1200px;
```
это не нужно, .page-header, как и все остальные блоки, просто имеет ширину, равную ширине контейнера, в котором лежит. то есть можно написать .page-header {width: 100%}, а можно вообще не писать ширину, у блочных элементов она по умолчанию равна 100%

talequale commented on styles/styles.css:
```diff
@@ -67,13 +78,45 @@ body {
+
+.navigation {
+  display: flex;
+  width: 1060px;
```
смысл не совсем в том, что навигация имеет ширину 1060px, а в том, что между навигацией и краями контейнера есть расстояние в 70px. это боковые паддинги навигации либо хедера

talequale commented on styles/styles.css:
```diff
@@ -88,34 +131,63 @@ body {
 }
 
 .welcome {
+  width: 1200px;
```
ширину можно не писать

talequale commented on styles/styles.css:
```diff
@@ -88,34 +131,63 @@ body {
+  display: block;
+  width: 458px;
+  margin: 0 auto;
+  padding-top: 51px;
+  padding-bottom: 82px;
```
эти стили висят на теге img, не совсем корректно писать ему паддинги, т к img это тег без контента, а паддинги это внутренние поля элемента, то есть то, что отодвигает контент элемента от его краёв

поэтому паддинги надо вешать на родителя картинки, а не на картинку

talequale commented on styles/styles.css:
```diff
@@ -88,34 +131,63 @@ body {
 .advantages-main-title {
+  width: 700px;
+  margin: 0 auto;
+  padding: 0;
```
у тегов заголовков дефолтных паддингов и так нет, можно не писать

talequale commented on styles/styles.css:
```diff
@@ -163,19 +261,26 @@ body {
 
 .subscribe {
   width: 1200px;
   height: 411px;
```
фиксированную высоту нельзя указывать блокам с контентом, который может меняться 
можно указать минимальную высоту, и точно нужно указать паддинги сверху и снизу (слева и справа тоже не повредит)

talequale commented on styles/styles.css:
```diff
@@ -190,6 +295,26 @@ body {
+.footer-container {
+  display: flex;
+  margin: 0 auto;
+  width: 1060px;
```
та же история, что и с навигацией, тут нужна не ширина, а паддинги по 70 пикселей по краям

talequale commented on styles/styles.css:
```diff
@@ -190,6 +295,26 @@ body {
+}
+
+.footer-social {
+  width: 142px;
```
кнопок может быть больше, чем 3, соответственно и ширина блока может быть больше, ведь пустое пространство там есть

talequale commented on catalog.html:
```diff
@@ -39,7 +39,7 @@
+    <main class="main-container main-inner">
       <header class="inner-header">
         <h1 class="inner-header-title">Гостиницы Седоны</h1>
         <ol class="breadcrumbs-list">
```
вообще это ul, тут явно не перечисление. путь в хлебных крошках может быть разный - либо это вертикальны путь от корня сайта, либо путь, по которому пользователь попал на эту страницу. я б не сказала, что тут есть какое-то подобие 1-2-3

OlgaNikl replied:
точно помню, что Андрей Серёдкин говорил на лайве, но не помню на каком, что хлебные крошки это ol, поэтому исправила.

OlgaNikl replied:
возможно я не правильно поняла

OlgaNikl replied:
наверное я неправильно поняла, исправлю

talequale replied:
может и говорил, получается, я с ним не согласна ) так что тут на твой выбор

talequale commented on styles/styles.css:
```diff
@@ -212,24 +345,51 @@ body {
+  display: flex;
+  flex-direction: column;
+  padding: 0;
+  width: 151px;
+  margin-right: 70px;
```
не обязательно впритык по ширине заголовка измерять ширину блока. текст же может быть другим. можно оставить небольшой маргин 20px, а остальное добавить к ширине (т е 200px)

talequale commented on styles/styles.css:
```diff
@@ -212,24 +345,51 @@ body {
+  margin-right: 70px;
+}
+
+.housing-type-filter {
+  width: 146px;
+}
+
+.housing-type-filter {
+  width: 146px;
+}
```
тут кажется задублировались стили. и на самом деле можно сделать .infrastructure-filter и .housing-type-filter одним классом вообще, т к ничто не мешает нам сделать их одинаковыми по ширине

talequale commented on styles/styles.css:
```diff
@@ -240,13 +400,38 @@ body {
 
+.catalog-view {
+  display: flex;
+  width: 1057px;
```
тоже паддинги нужны вместо ширины

talequale commented on styles/styles.css:
```diff
@@ -240,13 +400,38 @@ body {
+  margin: 0 auto;
+}
+
+.hotels-results {
+  min-width: 338px;
+  margin-right: 191px;
+}
```
![image](https://user-images.githubusercontent.com/11165551/173200154-a13296fd-f1be-410b-894a-87925b339ba7.png)
надо объединить два блока стилей для hotels-results

talequale commented on styles/styles.css:
```diff
@@ -240,13 +400,38 @@ body {
+
+.hotels-results {
+  min-width: 338px;
+  margin-right: 191px;
```
можно просто margin-right: auto

talequale commented on styles/styles.css:
```diff
@@ -240,13 +400,38 @@ body {
+  margin-right: 191px;
+}
+
+.select-control {
+  width: 290px;
+  margin-right: 78px;
+}
```
селект в пагинации тоже надо сделать тем же самым классом, потому что это одинаковые контролы. просто можно добавить дополнительный класс для указания другой ширины

talequale commented on styles/styles.css:
```diff
@@ -240,13 +400,38 @@ body {
+.hotels-list {
+  display: flex;
+  flex-wrap: wrap;
+  width: 1056px;
```
тоже паддинги, а не ширина

talequale reviewed (commented):
![HTML Academy: Седона 2022-06-11 21-14-22](https://user-images.githubusercontent.com/11165551/173200086-c289f668-47ba-40e1-a977-31ed932815f1.jpg)
тут просто нужно обёртку кнопкам добавить, чтоб изолировать от родительского флекс-контейнера, который щас кнопки растягивает

pagination-view тоже надо построить сетку (паддинги, флекс)

OlgaNikl committed:
исправлены недочеты

Переименован контейнер для body в каталоге; breadcrumbs-list изменен на ul; фильтрам отелей добавлен общий класс;  кнопки в фильтре обернуты в див чтобы не растягивались;  исправлены опечатки description; добавлены паддинги вместо ширины у секций;  добавлены стили для секции подписка на внутренней странице; добавлены стили пагинации; общий класс для селектов.

talequale commented on styles/styles.css:
```diff
@@ -272,6 +447,21 @@ body {
+.pagination-view {
+  display: flex;
+  align-items: center;
+  width: 1056px;
+  margin: 0 auto;
```
ширина как у всех блоков 100%. и не маргины по бокам, а паддинги по 70px, как и везде

talequale commented on styles/styles.css:
```diff
@@ -272,6 +447,21 @@ body {
+
+.pagination {
+  display: flex;
+  justify-content: space-between;
```
это неправильный подход (в этом можно убедиться, добавив ещё пару элементов в список). при заданной ширине контейнера элементам надо добавить макетные маргины просто
